e2store/src/utils.rs

```diff
@@ -47,3 +47,54 @@ pub fn underlying_io_error_kind(error: &Error) -> Option<io::ErrorKind> {
     }
     None
 }
+
+const ERA_DIR_URL: &str = "https://mainnet.era.nimbus.team/";
+
+/// Fetches era file download links
+pub async fn get_era_file_download_links(
+    http_client: &Client,
+) -> anyhow::Result<HashMap<u64, String>> {
+    let index_html = http_client
+        .get(ERA_DIR_URL)
+        .recv_string()
+        .await
+        .map_err(|e| anyhow!("{e}"))?;
+    let index_html = Html::parse_document(&index_html);
+    let selector = Selector::parse("a[href*='mainnet-']").expect("to be able to parse selector");
+    let era_files: HashMap<u64, String> = index_html
+        .select(&selector)
+        .map(|element| {
+            let href = element
+                .value()
+                .attr("href")
+                .expect("to be able to get href");
+            let epoch_index = href
+                .split('-')
+                .nth(1)
+                .expect("to be able to get epoch")
+                .parse::<u64>()
+                .expect("to be able to parse epoch");
+            (epoch_index, format!("{ERA_DIR_URL}{href}"))
+        })
+        .collect();
+    Ok(era_files)
```

Comment:
I am not sure what you mean do you want me to loop from 0 to max era count and see if all exist?

Because since we are using a hashmap, the the values are stored in the positions of the hash of the keys which wouldn't be in sorted order, if we were to iterate over the buckets in order

Reply:
I think loop and checking is simple enough. Something like:
```rust
ensure!(
    (0..(era_files.len()).all(|epoch| era_files.contains(&epoch)),
    "Epoch indices are not starting from zero or not consecutive",
);
```
You can also calculate min and max and check if they are as desired (this still requires looping, so no saving performance wise).